core/src/main/java/hudson/model/Run.java

```diff
@@ -2302,6 +2302,9 @@ public EnvVars getEnvironment() throws IOException, InterruptedException {
         for (EnvironmentContributor ec : EnvironmentContributor.all().reverseView())
             ec.buildEnvironmentFor(this,env,listener);
 
+        for (EnvironmentContributingAction a : getActions(EnvironmentContributingAction.class))
+            a.buildEnvVars(this,env,n);
```

Comment:
From my personal perspective some more generic way to do it is needed. I.e. some decorator that wraps all calls by try/catch. Or at least forces developers to do it.